src/common/Channel.hpp

```diff
@@ -122,6 +124,9 @@ class Channel : public std::enable_shared_from_this<Channel>, public MessageSink
 
     MessageSinkTraits sinkTraits() const final;
 
+    EmoteHolder *emotes();
+    const EmoteHolder *emotes() const;
```

Comment:
I'd prefer if emote logic was constrained to `TwitchChannel` since that's where all the current emote logic lives